packages/client/src/runtime/utils/createErrorMessageWithContext.ts

```diff
@@ -76,8 +76,8 @@ function getTemplateParameters(
 
   const contextFirstLine = Math.max(1, callLocation.lineNumber - 3)
   let source = SourceFileSlice.read(callLocation.fileName)?.slice(contextFirstLine, callLocation.lineNumber)
-  if (source) {
-    const invocationLine = source.lineAt(callLocation.lineNumber)
+  const invocationLine = source?.lineAt(callLocation.lineNumber)
```

Comment:
Let's also update the signature of `lineAt` to explicitly specify `string | undefined` as the return type. Otherwise the issue may be accidentally introduced back when moving the code around.

Reply:
Good point, done.